### ISSUE TITLE
Parsing error of valid HTML document

### ISSUE THREAD

**Issue by dimanech** [FabiowQuixada/isml-linter#26]
Hi @FabiowQuixada , I try to start with isml-linter, but stuck with parsing error with valid HTML. Could you please help with it?
error message: "An unexpected error happened while parsing the following templates:"
node v14.17.0
file content:
```
<!DOCTYPE html>
<head>
    <meta charset="UTF-8"/>
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Mail factory</title>

    <style>
        :root {
            --color-blue: #3466e5;
        }

        html {
            margin: 0 20px;
        }

        body {
            margin: 0 auto;
            max-width: 800px;
            font: normal 16px/1.2 Arial, sans-serif;
        }

        table {
            border-collapse: collapse;
            margin: 0 -10px 10px;
            border-bottom: 1px solid grey;
            width: 100%;
        }

        th,
        td {
            padding: 5px 10px;
        }

        th {
            padding: 0 10px;
            text-align: left;
        }

        thead th {
            border-bottom: 1px solid grey;
            padding-bottom: 10px;
        }

        tbody tr:first-child td {
            padding-top: 15px;
        }

        tbody tr:last-child td {
            padding-bottom: 20px;
        }

        input {
            display: inline-block;
            min-height: 22px;
            border: 1px solid grey;
            font-size: 14px;
        }

        input.m-mail {
            width: 230px;
        }

        .b-button {
            appearance: none;
            display: inline-block;
            min-height: 22px;
            line-height: 22px;
            text-decoration: none;
            color: white;
            background-color: var(--color-blue);
            border: 2px solid var(--color-blue);
            font-size: inherit;
            margin-right: 5px;
            padding: 0 10px;
            cursor: pointer;
            user-select: none;
        }

        .b-button.m-outlined {
            background-color: transparent;
            color: var(--color-blue);
        }

        .b-button.m-outlined[disabled] {
            color: #9e9e9e;
            border-color: #bbb;
            pointer-events: none;
            cursor: default;
        }

        .b-form {
            margin: 40px auto 20px;
        }

        .b-legend {
            color: grey;
            font-size: 14px;
            margin: 5px 0;
        }

        .b-alert {
            border: 1px solid #eee;
            margin: 10px 0;
            text-align: center;
            padding: 10px;
        }

        .b-alert.m-warn {
            background: repeating-linear-gradient(135deg, yellow 0px, yellow 15px, #eee 15px, #eee 30px);
        }

        .b-alert.m-error {
            border-color: red;
            background: red;
            color: white;
        }

        .b-alert.m-success {
            border-color: greenyellow;
            background-color: greenyellow;
        }

        .b-address {
            display: block;
            padding: 10px 0;
            font-weight: bold;
        }

        .b-logo {
            display: block;
            margin: 40px auto 60px;
            text-align: center;
            position: relative;
            max-width: 280px;
        }

        .b-logo.m-active .b-logo-windows {
            fill: var(--color-blue);
        }

        .b-logo-smoke {
            display: none;
        }

        .b-logo.m-active .b-logo-smoke {
            position: absolute;
            display: block;
            border: 1.8px solid black;
            border-radius: 20px;
            background-color: white;
            animation: smoke 14s ease-in-out infinite;
            z-index: -1;
        }

        .b-logo-smoke.m-1 {
            width: 5px;
            height: 5px;
            right: 29px;
            top: 1px;
        }

        .b-logo-smoke.m-2 {
            animation-delay: 0.3s;
            width: 3px;
            height: 3px;
            right: 30px;
            top: 1px;
        }

        .b-logo-smoke.m-3 {
            animation-delay: 0.5s;
            width: 3px;
            height: 3px;
            right: 14px;
            top: -4px;
        }


        .b-logo-smoke.m-4 {
            animation-delay: 0.8s;
            width: 2px;
            height: 2px;
            right: 13px;
            top: -4px;
        }

        @keyframes smoke {
            0% {
                opacity: 0;
            }
            5% {
                opacity: 1;
            }
            40% {
                opacity: 0;
                top: -10px;
                right: 60px;
                transform: scale(1.8);
            }
            100% {
                opacity: 0;
            }
        }
    </style>
</head>
<body>
    <a class="b-logo" id="logo" href="">
        <svg xmlns="http://www.w3.org/2000/svg"
             viewBox="0 0 170.876 29.023"
             width="280">
            <g stroke-width="4.373" fill="black">
                <path class="b-logo-windows"
                      d="M143.285 22.266h9.68v.948h-9.68zM154.912 22.256h.948v2.904h-.948zM157.817 22.256h.948v2.904h-.948zM160.721 22.256h.948v2.904h-.948zM163.625 22.256h.948v2.904h-.948zM166.53 22.256h.948v2.904h-.948z"/>
                <path d="M170.382 28.075v-14.33l-6.818 3.408L162.611 0h-4.766l-.978 15.662-2.996 1.498-.949-12.804h-4.747l-.983 11.302-6.801 3.401v9.016h-.494v.948h30.979v-.948zM158.736.948h2.979l.055.988h-3.096zm-.121 1.936h3.207l.055.988h-3.324zm-.121 1.936h3.436l.71 12.795-1.94.97v-4.84l-2.853 1.427zm-9.45.484h2.998l.073.989h-3.157zm-.168 1.936h3.31l.073.989h-3.47zm-.169 1.937h3.622l.625 8.441-1.934.967v-4.84l-2.833 1.416zm-5.412 18.898h-1.956v-.988h1.956v.988zm8.713 0h-7.765v-3.409h7.765zm17.426 0h-16.478v-.988h16.478zm0-1.936h-16.478v-2.42h-9.66v2.42h-1.957v-6.494l8.733-4.367v4.84l9.68-4.84v4.84l9.682-4.84v10.86z"/>
            </g>
            <g aria-label="mail.factory">
                <path d="M0 23.815h3.416v-7.393c0-1.363.865-2.269 2.004-2.269 1.122 0 1.876.77 1.876 1.98v7.682h3.311v-7.521c0-1.275.73-2.14 1.973-2.14 1.09 0 1.908.68 1.908 2.052v7.609h3.407v-8.283c0-2.67-1.587-4.193-3.88-4.193-1.804 0-3.207.922-3.712 2.333h-.129c-.392-1.427-1.651-2.333-3.343-2.333-1.66 0-2.918.882-3.432 2.333h-.144V11.5H0zM24.101 24.047c1.82 0 2.999-.794 3.6-1.94h.096v1.708h3.24v-8.307c0-2.934-2.486-4.169-5.228-4.169-2.95 0-4.89 1.411-5.364 3.656l3.159.257c.233-.818.962-1.42 2.189-1.42 1.162 0 1.828.586 1.828 1.596v.048c0 .794-.842.898-2.983 1.107-2.437.224-4.626 1.042-4.626 3.792 0 2.437 1.74 3.672 4.09 3.672zm.978-2.357c-1.05 0-1.804-.49-1.804-1.427 0-.962.794-1.435 1.997-1.604.745-.104 1.964-.28 2.373-.553v1.307c0 1.29-1.066 2.277-2.566 2.277zM33.69 23.815h3.416V11.499H33.69zm1.716-13.903c1.018 0 1.852-.778 1.852-1.732 0-.946-.834-1.724-1.852-1.724-1.01 0-1.844.778-1.844 1.724 0 .954.834 1.732 1.844 1.732zM43.247 7.394h-3.415v16.42h3.415z"/>
                <path fill="#3466e5"
                      d="M47.97 24.023c1.026 0 1.916-.858 1.924-1.924-.008-1.05-.898-1.908-1.924-1.908-1.059 0-1.933.857-1.925 1.908a1.92 1.92 0 001.925 1.924z"/>
                <path d="M59.315 11.5h-2.446v-.826c0-.834.337-1.323 1.34-1.323.408 0 .825.088 1.098.176l.601-2.566a8.996 8.996 0 00-2.413-.336c-2.317 0-4.041 1.306-4.041 3.984v.89h-1.74v2.566h1.74v9.75h3.415v-9.75h2.446zM64.462 24.047c1.82 0 2.999-.794 3.6-1.94h.096v1.708h3.24v-8.307c0-2.934-2.486-4.169-5.228-4.169-2.95 0-4.891 1.411-5.364 3.656l3.159.257c.232-.818.962-1.42 2.189-1.42 1.162 0 1.828.586 1.828 1.596v.048c0 .794-.842.898-2.983 1.107-2.437.224-4.626 1.042-4.626 3.792 0 2.437 1.74 3.672 4.089 3.672zm.978-2.357c-1.05 0-1.804-.49-1.804-1.427 0-.962.794-1.435 1.997-1.604.745-.104 1.964-.28 2.373-.553v1.307c0 1.29-1.066 2.277-2.566 2.277zM79.615 24.055c3.28 0 5.34-1.924 5.5-4.754h-3.223c-.2 1.314-1.066 2.052-2.236 2.052-1.596 0-2.63-1.339-2.63-3.696 0-2.325 1.042-3.656 2.63-3.656 1.25 0 2.052.826 2.236 2.052h3.224c-.145-2.846-2.301-4.714-5.517-4.714-3.736 0-6.045 2.59-6.045 6.366 0 3.744 2.27 6.35 6.061 6.35zM93.863 11.5h-2.317V8.548H88.13v2.95h-1.684v2.566h1.684v6.414c-.016 2.414 1.628 3.608 4.105 3.504a6.645 6.645 0 001.852-.32l-.537-2.542c-.168.032-.529.112-.85.112-.681 0-1.154-.257-1.154-1.203v-5.965h2.317zM101.52 24.055c3.736 0 6.061-2.558 6.061-6.35 0-3.816-2.325-6.366-6.061-6.366-3.737 0-6.062 2.55-6.062 6.366 0 3.792 2.325 6.35 6.062 6.35zm.016-2.646c-1.724 0-2.606-1.58-2.606-3.728 0-2.149.882-3.736 2.606-3.736 1.691 0 2.573 1.587 2.573 3.736s-.882 3.728-2.573 3.728zM109.81 23.815h3.416v-6.968c0-1.515 1.106-2.557 2.613-2.557.473 0 1.123.08 1.443.184v-3.03a5.124 5.124 0 00-1.074-.121c-1.379 0-2.51.802-2.958 2.325h-.129V11.5h-3.311zM121.532 28.433c2.541 0 3.888-1.3 4.586-3.288l4.786-13.646h-3.616l-2.573 9.044h-.128l-2.55-9.044h-3.592l4.418 12.668-.2.521c-.45 1.155-1.308 1.211-2.51.842l-.77 2.55c.49.208 1.283.353 2.149.353z"/>
            </g>
        </svg>
        <div class="b-logo-smoke m-2"></div>
        <div class="b-logo-smoke m-4"></div>
        <div class="b-logo-smoke m-3"></div>
        <div class="b-logo-smoke m-1"></div>
    </a>

    <div class="b-alert m-warn" id="local-server-check" hidden>
        You are not into
        <strong>develop mode</strong>! To use hot reload capabilities run `npm run dev:mails:watch`
    </div>
    <script>
        const showErrorMessage = () => {
            document.getElementById('local-server-check').removeAttribute('hidden');
            document.querySelectorAll('.b-button.m-develop').forEach(button => button.setAttribute('disabled', 'disabled'));
        };

        fetch('http://127.0.0.1:3001/mails.css').then(response => {
            if (!response.ok) {
                showErrorMessage();
            } else {
                document.getElementById('logo').classList.add('m-active');
            }
        }).catch(showErrorMessage);
    </script>
    <form
        class="b-form"
        method="post"
        action="${URLUtils.url('MailFactory-Send')}"
    >
        <table>
            <thead>
                <tr>
                    <th>Mail</th>
                    <th>Action</th>
                </tr>
            </thead>
            <tbody>
                <isloop items="${pdict.mails}" var="mail">
                    <tr>
                        <td>${mail}</td>
                        <td>
                            <a
                                class="b-button m-outlined m-develop"
                                data-mail="${mail}"
                                data-href="${URLUtils.url('MailFactory-Preview', 'mid', mail, 'useMailStyles', 'true')}"
                                href="${URLUtils.url('MailFactory-Preview', 'mid', mail, 'useMailStyles', 'true')}"
                            >
                                Develop
                            </a>
                            <a
                                class="b-button m-outlined"
                                data-mail="${mail}"
                                data-href="${URLUtils.url('MailFactory-Preview', 'mid', mail)}"
                                href="${URLUtils.url('MailFactory-Preview', 'mid', mail)}"
                            >
                                Preview
                            </a>
                            <isif condition="${mail === 'orderConfirmation'}">
                                <input
                                    id="orderIDinput"
                                    type="text"
                                    placeholder="orderID"
                                    name="orderID"
                                    oninput="onOrderInput(this.value)"
                                />
                            </isif>
                            <isif condition="${mail === 'backInStockSubscription'}">
                                <input
                                    id="productIDinput"
                                    type="text"
                                    placeholder="productID"
                                    name="productID"
                                    oninput="onProductInput(this.value)"
                                />
                            </isif>
                            <button
                                class="b-button"
                                data-mailid="${mail}"
                                name="mailid"
                                value="${mail}"
                            >
                                Send
                            </button>
                        </td>
                    </tr>
                </isloop>
            </tbody>
        </table>

        <label class="b-address">
            Send to:
            <input
                class="m-mail"
                type="email"
                required
                placeholder="email address"
                value="${pdict.recipient}"
                name="mailTo"
            />
        </label>
    </form>

    <p class="b-legend">
        <strong>Develop</strong> - Use CSS classes and livereload. You need to run local server to use this.
    </p>
    <p class="b-legend">
        <strong>Preview</strong> - Use inline styles. Same as would be send to user.
    </p>

    <script>
        function onOrderInput(orderID) {
            document.querySelectorAll('[data-mail="orderConfirmation"]').forEach((link) => {
                if (orderID === '') {
                    link.setAttribute('disabled', 'disabled');
                } else {
                    link.setAttribute('href', link.getAttribute('data-href') + '&orderID=' + orderID);
                    link.removeAttribute('disabled');
                }
            });
        }
        // input autocomplete initialized with delay
        setTimeout(() => onOrderInput(document.getElementById('orderIDinput').value), 100);

        function onProductInput(productID) {
            document.querySelectorAll('[data-mail="backInStockSubscription"]').forEach((link) => {
                link.setAttribute('href', link.getAttribute('data-href') + '&productID=' + productID);
            });
        }
    </script>

</body>
```

**Comment by FabiowQuixada** [comment]
hey @dimanech , thanks for reaching out!

I've taken a look at this, and the fix is available in v5.35.2, please update ISML Linter and let me know if it works on your side.

Thanks!

**Comment by dimanech** [comment]
@FabiowQuixada It works now! Thank you for the quick fix! =)

**Comment by FabiowQuixada** [comment]
Glad to hear! :D
